src/doc.d

```diff
@@ -2151,8 +2152,34 @@ extern (C++) void highlightText(Scope* sc, Dsymbols* a, OutBuffer* buf, size_t o
                 static __gshared const(char)* blankline = "$(DDOC_BLANKLINE)\n";
                 i = buf.insert(i, blankline, strlen(blankline));
             }
-            leadingBlank = 1;
             iLineStart = i + 1;
+            if (!leadingBlank)
+            {
+                leadingBlank = true;
+            }
+            else if (!inCode)
+            {
+                // Group two or more newlines into a paragraph break
+                auto scout = iLineStart;
+                for (; scout < buf.offset; ++scout)
+                {
+                    c = buf.data[scout];
+                    if (c == ' ' || c == '\t')
+                    {
+                        continue;
+                    }
+                    if (c == '\n')
+                    {
+                        iLineStart = scout + 1;
```

Comment:
This is probably what is breaking the autotester, since iLineStart is also used for the DDOC_BLANKLINE test up on line 2150: https://github.com/andralex/dmd/blob/DDOC_PARAGRAPH_SEPARATOR/src/doc.d#L2150